lib/pry-doc/pry_ext/show_source_with_c_internals/c_file.rb

```diff
@@ -17,7 +17,7 @@ class CFile
       attr_accessor :file_name
       attr_reader :ruby_source_folder
 
-      def initialize(file_name:, content:, ruby_source_folder: nil)
+      def initialize(file_name: nil, content: nil, ruby_source_folder: nil)
```

Comment:
If kwargs are supported on MRI 2.1+, then how does this work on 2.0? This is also kwargs, right?

Reply:
> If kwargs are supported on MRI 2.1+

No.  You seem to be confusing **Keyword argument** and **Required keyword argument**

Keyword argument available since MRI 2.0+
https://www.ruby-lang.org/en/news/2013/02/24/ruby-2-0-0-p0-is-released/

But required keyword argument available since MRI 2.1+
https://github.com/ruby/ruby/blob/v2_1_0/NEWS#language-changes

`def initialize(file_name:, content:, ruby_source_folder: nil)` is required keyword argument syntax (for MRI 2.1+)

But `def initialize(file_name: nil, content: nil, ruby_source_folder: nil)` is (non required) keyword argument syntax (for MRI 2.0+)

See following for details.
https://robots.thoughtbot.com/ruby-2-keyword-arguments#required-keyword-arguments

Reply:
Good points, makes sense, thanks.